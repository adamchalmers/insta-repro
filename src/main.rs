#[derive(serde::Serialize, serde::Deserialize)]
struct Node {
    nodes: Vec<Node>,
    data: f64,
}

fn main() {}

#[test]
fn test_recurse() {
    let data = Node {
        nodes: vec![
            Node {
                nodes: vec![Node {
                    nodes: vec![],
                    data: f64::tan(4.0),
                }],
                data: f64::tan(2.0),
            },
            Node {
                nodes: vec![],
                data: f64::tan(3.0),
            },
        ],
        data: f64::tan(1.0),
    };
    insta::assert_yaml_snapshot!("data", data, {
        ".**[]" => insta::rounded_redaction(3),
        ".**" => insta::rounded_redaction(3),
    });
}
