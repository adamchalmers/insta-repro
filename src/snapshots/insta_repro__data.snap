---
source: src/main.rs
expression: data
snapshot_kind: text
---
nodes:
  - nodes:
      - nodes: []
        data: 1.1578212823495775
    data: -2.185039863261519
  - nodes: []
    data: -0.1425465430742778
data: 1.557
